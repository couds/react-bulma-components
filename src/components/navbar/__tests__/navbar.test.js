import { JSDOM } from 'jsdom';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { render } from 'react-testing-library';
import { getHtmlClasses, Navbar } from '..';

describe('Navbar component', () => {
  beforeEach(() => {
    // eslint-disable-next-line
    window = new JSDOM('<!doctype html><html><body><div id="app-root"></div></body></html>').window;

    global.window = window;
    global.document = window.document;
    global.navigator = {
      userAgent: 'node.js'
    };
  });

  it.each([
    [Navbar],
    [Navbar.Brand],
    [Navbar.Burger],
    [Navbar.Container],
    [Navbar.Divider],
    [Navbar.Dropdown],
    [Navbar.Item],
    [Navbar.Link],
    [Navbar.Menu]
  ])('should render', Component => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render on server side', () => {
    const html = renderToString(<Navbar fixed="top" />);
    expect(html).toMatchSnapshot();
    expect(getHtmlClasses()).toMatchSnapshot('top');
  });

  it('should have Navbar classname', () => {
    const { asFragment, unmount } = render(
      <Navbar>
        <Navbar.Brand>
          <Navbar.Item renderAs="a" href="#">
            <img src="/" alt="" />
          </Navbar.Item>
          <Navbar.Burger className="trigger-menu" />
        </Navbar.Brand>
        <div>TEST</div>
        <Navbar.Menu>
          <Navbar.Container>
            <Navbar.Item dropdown hoverable>
              <Navbar.Link>Docs</Navbar.Link>
              <Navbar.Dropdown boxed>
                <Navbar.Item href="#">Home</Navbar.Item>
                <Navbar.Divider />
                <Navbar.Item href="#">Home</Navbar.Item>
              </Navbar.Dropdown>
            </Navbar.Item>
          </Navbar.Container>
        </Navbar.Menu>
      </Navbar>
    );
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
  it('should concat Bulma class with classes in props', () => {
    const { asFragment, unmount } = render(<Navbar className="other-class test" />);
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
  it('should render as an html section', () => {
    const { asFragment, unmount } = render(<Navbar renderAs="section" />);
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
  it('should have custom inline styles', () => {
    const { asFragment, unmount } = render(<Navbar style={{ width: 200, zIndex: 1 }} />);
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });

  it('should be fixed on top', () => {
    const { asFragment, unmount } = render(<Navbar fixed="top" />);
    expect(asFragment()).toMatchSnapshot();
    unmount();
  });
});
