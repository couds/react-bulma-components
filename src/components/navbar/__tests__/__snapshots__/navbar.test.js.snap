// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should be fixed on top 1`] = `
<DocumentFragment>
  <nav
    class="navbar is-fixed-top"
    role="navigation"
  />
</DocumentFragment>
`;

exports[`Navbar component should concat Bulma class with classes in props 1`] = `
<DocumentFragment>
  <nav
    class="navbar other-class test"
    role="navigation"
  />
</DocumentFragment>
`;

exports[`Navbar component should have Navbar classname 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="#"
      >
        <img
          alt=""
          src="/"
        />
      </a>
      <div
        class="navbar-burger trigger-menu"
        role="button"
        style="outline: none;"
        tabindex="0"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
    <div>
      TEST
    </div>
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
      >
        <span
          class="navbar-item has-dropdown is-hoverable"
        >
          <span
            class="navbar-link"
          >
            Docs
          </span>
          <span
            class="navbar-dropdown is-boxed"
          >
            <a
              class="navbar-item"
              href="#"
            >
              Home
            </a>
            <div
              class="navbar-divider"
            />
            <a
              class="navbar-item"
              href="#"
            >
              Home
            </a>
          </span>
        </span>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`Navbar component should have custom inline styles 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    role="navigation"
    style="width: 200px; z-index: 1;"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    role="navigation"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 2`] = `
<DocumentFragment>
  <div
    class="navbar-brand"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 3`] = `
<DocumentFragment>
  <div
    class="navbar-burger is-active"
    role="button"
    style="outline: none;"
    tabindex="0"
  >
    <span />
    <span />
    <span />
  </div>
</DocumentFragment>
`;

exports[`Navbar component should render 4`] = `
<DocumentFragment>
  <div
    class="navbar-start"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 5`] = `
<DocumentFragment>
  <div
    class="navbar-divider"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 6`] = `
<DocumentFragment>
  <span
    class="navbar-dropdown"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 7`] = `
<DocumentFragment>
  <a
    class="navbar-item"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 8`] = `
<DocumentFragment>
  <span
    class="navbar-link"
  />
</DocumentFragment>
`;

exports[`Navbar component should render 9`] = `
<DocumentFragment>
  <div
    class="navbar-menu is-active"
  />
</DocumentFragment>
`;

exports[`Navbar component should render as an html section 1`] = `
<DocumentFragment>
  <section
    class="navbar"
    role="navigation"
  />
</DocumentFragment>
`;

exports[`Navbar component should render on server side 1`] = `"<nav class=\\"navbar is-fixed-top\\" role=\\"navigation\\"></nav>"`;

exports[`Navbar component should render on server side: top 1`] = `""`;
